### PR TITLE
Fix: convert ZClientViewController

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?
Convert `ZClientViewController` to Swift.
Created new `MockSelfUser` class, a minimal mock class to conform `SelfLegalHoldSubject`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 1]
```diff
@@ -19,12 +19,467 @@
...
+    ///
+    /// This method should only be called when the list selection changes, or internally by other zclientviewcontroller
+    /// methods.
+    /// - Parameters:
+    ///   - conversation: <#conversation description#>
+    ///   - message: <#message description#>
+    ///   - focus: <#focus description#>
+    ///   - animated: <#animated description#>
+    /// - Returns: return YES if it actually switched views, NO if nothing changed (ie: we were already looking at the conversation)
+//    func load(_ conversation: ZMConversation?, scrollTo message: ZMConversationMessage?, focusOnView focus: Bool, animated: Bool) -> Bool {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: Delete?

File: Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 2]
```diff
@@ -19,12 +19,467 @@
...
+        }
+    }
+    
+    
+    /// init method for testing allows injecting an Account object and self user
+    ///
+    /// - Parameters:
+    ///   - account: an Account object
+    ///   - selfUser: a SelfUserType object
+    public convenience init(account: Account,
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: Isn't this actually a required initialiser? I would then remove the `convenience` attribute and update the method description.

File: Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 3]
```diff
@@ -19,12 +19,467 @@
...
+    private(set) var conversationRootViewController: UIViewController?
+    private(set) var currentConversation: ZMConversation?
+    var isComingFromRegistration = false
+    var needToShowDataUsagePermissionDialog = false
+    let wireSplitViewController: SplitViewController = SplitViewController()
     
-    // MARK: - Adressbook Upload
+    @objc
+    private(set) var mediaPlaybackManager: MediaPlaybackManager?
+    private(set) var conversationListViewController: ConversationListViewController!
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: Does this really need be force unwrapped?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: changed to non-optional

**Pull request merged**